public/pages/Snapshots/components/RestoreActivitiesPanel/RestoreActivitiesPanel.tsx

```diff
@@ -102,33 +144,48 @@ export const RestoreActivitiesPanel = ({ snapshotManagementService, snapshotId,
       ) {
         const info = response[responseItem].shards[0];
         const stage = stages.indexOf(info.stage);
-        const size = `${(info.index.size.total_in_bytes / 1024 ** 2).toFixed(2)}mb`;
-
         const time = {
           start_time: info.start_time_in_millis,
           stop_time: info.stop_time_in_millis ? info.stop_time_in_millis : Date.now()
         };
 
-        doneCount = stage === 4 ? doneCount + 1 : doneCount;
+        doneCount = stage === lastStage ? doneCount + 1 : doneCount;
         stageIndex = stage < stageIndex ? stage : stageIndex;
 
         maxStopTime = maxStopTime && maxStopTime > time.stop_time ? maxStopTime : time.stop_time;
 
+        const indexStatus = modifiedStages[info.stage as keyof ModifiedStages];
+
         if (info.source.index && info.source.snapshot === snapshotId) {
           minStartTime = minStartTime && minStartTime < time.start_time ? minStartTime : time.start_time;
-          indexes.push({ index: info.source.index, "store.size": size });
+          indexes.push({ index: info.source.index, "restore_status": indexStatus });
         }
       }
     }
-    let percent = Math.floor((doneCount / restoreCount) * 100);
 
-    setIndices(indexes);
+    const updatedIndices = [...indexes];
+    const indicesStarted = indexes.map((index) => index.index);
+
+    for (let index of indicesToRestore) {
+      if (indicesStarted.indexOf(index) < 0) {
+        updatedIndices.push({ index, restore_status: "Pending" })
+      }
+    }
+
+    const sortedUpdatedIndices = updatedIndices.sort((a, b) => {
+      if (a.index < b.index) return -1;
+      if (a.index > b.index) return 1;
+      return 0;
```

Comment:
Can be written as `return a.index - b.index`

Reply:
I'll have a look and change if it works.  good point.